repository_service_tuf_worker/repository.py

```diff
@@ -61,7 +61,13 @@ class Roles(enum.Enum):
 
     @staticmethod
     def online_roles() -> List[str]:
-        return [Targets.type, Snapshot.type, Timestamp.type, "bins"]
+        online_roles = [Targets.type, Snapshot.type, Timestamp.type, "bins"]
+        settings_repository = get_worker_settings()
+        settings_repository.reload()
+        if not settings_repository.get_fresh("TARGETS_ONLINE_KEY", True):
+            online_roles = [Snapshot.type, Timestamp.type, "bins"]
```

Comment:
Shouldn't bins be excluded as well?
I might be missing some details from the workflow - if so, please ignore to not waste time with the comment. 
Otherwise, given that bins are delegates and `TARGETS_ONLINE_KEY` is `False`, it looks incorrect to have them in the online roles.

Reply:
Bins could use the online key. 
Even though targets could be offline in the future this doesn't mean we are required to use offline keys for all its delegations. The security properties coming from targets utilizing an offline key wouldn't be lost as when adding, removing or downloading artifact information all requires going through `targets` first which will be cryptographically signed by an offline key.